gaurun/apns_http2.go

```diff
@@ -36,20 +43,46 @@ func NewTransportHttp2(cert tls.Certificate) (*http.Transport, error) {
 	return transport, nil
 }
 
-func NewApnsClientHttp2(certPath, keyPath, keyPassphrase string) (*http.Client, error) {
+func NewApnsClientHttp2(certPath, keyPath, keyPassphrase string) (APNsClient, error) {
 	cert, err := loadX509KeyPairWithPassword(certPath, keyPath, keyPassphrase)
 	if err != nil {
-		return nil, err
+		return APNsClient{}, err
 	}
 
 	transport, err := NewTransportHttp2(cert)
 	if err != nil {
-		return nil, err
+		return APNsClient{}, err
 	}
 
-	return &http.Client{
-		Transport: transport,
-		Timeout:   time.Duration(ConfGaurun.Ios.Timeout) * time.Second,
+	return APNsClient{
+		HTTPClient: &http.Client{
+			Transport: transport,
+			Timeout:   time.Duration(ConfGaurun.Ios.Timeout) * time.Second,
+		},
+	}, nil
+}
+
+func NewApnsClientHttp2ForToken(authKey *ecdsa.PrivateKey, keyID, teamID string) (APNsClient, error) {
+	authToken := &token.Token{
+		AuthKey: authKey,
+		KeyID:   keyID,
+		TeamID:  teamID,
+	}
+
+	return APNsClient{
+		HTTPClient: &http.Client{
+			Transport: &http.Transport{
+				MaxIdleConnsPerHost: ConfGaurun.Ios.KeepAliveConns,
+				Dial: (&net.Dialer{
+					Timeout:   time.Duration(ConfGaurun.Ios.Timeout) * time.Second,
+					KeepAlive: time.Duration(keepAliveInterval(ConfGaurun.Ios.KeepAliveTimeout)) * time.Second,
+				}).Dial,
+				IdleConnTimeout:   time.Duration(ConfGaurun.Ios.KeepAliveTimeout) * time.Second,
+				ForceAttemptHTTP2: true,
+			},
+			Timeout: time.Duration(ConfGaurun.Ios.Timeout) * time.Second,
+		},
+		Token: authToken,
```

Comment:
Nest-level is deep. With assigning transport to variable beforehand, make shallow a little more.

Reply:
I fixed it.